include/bluetooth/att.h

```diff
@@ -59,12 +59,53 @@ extern "C" {
 #if defined(CONFIG_BT_TESTING)
 
 int bt_eatt_disconnect_one(struct bt_conn *conn);
-int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
 size_t bt_eatt_count(struct bt_conn *conn);
 
 #endif /* CONFIG_BT_TESTING */
+
+/** @brief Connect Enhanced ATT channels
+ *
+ * Sends a series of Credit Based Connection Requests to connect @p num_channels
+ * Enhanced ATT channels. The peer may have limited resources and fewer channels
+ * may be created.
+ *
+ * @param conn The connection to send the request on
+ * @param num_channels The number of Enhanced ATT beaeres to request.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels);
+
 #endif /* CONFIG_BT_EATT */
 
+/* ATT channel information */
+struct bt_att_chan_info {
+	/* The connection the ATT channel belongs to */
+	struct bt_conn *conn;
+	/* The Channel ID of the ATT channel */
+	uint16_t cid;
+	/* The Maximum Transmission Unit of the ATT channel */
+	uint16_t mtu;
+	/* The Maximum PDU payload Size of the ATT channel */
+	uint16_t mps;
+};
+
+/** @brief ATT callback structure. */
+struct bt_eatt_cb {
+	/** @brief An ATT channel has been connected. */
```

Comment:
Yes, there currently is no API to control each channel, so the CID isn't very useful.

The MTU may be useful because the MTU of the channel may be smaller than what is configured, as the peer may not have enough resources.

I am open to adding/removing things from this struct if they are not useful. We can add more later on if needed. Is there other information about the EATT channel that may be useful?

Reply:
Let's keep it as minimum as possible and then expand on it as necessary.

Reply:
Yes :) Then we'll just add what is requested as it goes on.